src/ucp/core/ucp_request.c

```diff
@@ -614,7 +617,13 @@ void ucp_request_send_state_ff(ucp_request_t *req, ucs_status_t status)
             req->send.state.uct_comp.func(&req->send.state.uct_comp);
         }
     } else {
+        ep        = req->send.ep;
+        super_req = (req->super_req != NULL) ? ucp_request_get_super(req) : NULL;
         ucp_request_complete_send(req, status);
+
+        if (super_req != NULL) {
+            ucp_ep_req_purge(ep, super_req, status, 1);
+        }
```

Comment:
nice catch! fix seems weird thought:
1. seems not good to mix purge and ff
2. how can we assume super_req is valid (or NULL) if we don't know anything about the request? it could be user_data
3. for rdnv request, isn't uct_comp.func!=NULL? what is this case handling? maybe need special case (like we do for flush)?